src/core/Restriction.ml

```diff
@@ -5,12 +5,18 @@ type dim = I.t
 
 type eqn = dim * dim
 
-module UF = DisjointSet.Make (PersistentTable.M)
```

Comment:
@favonia I don't know if the changes in this file are useful at all. should we revert or keep them?